libs/core/langchain_core/messages/ai.py

```diff
@@ -16,13 +29,36 @@ class AIMessage(BaseMessage):
         conversation.
     """
 
+    tool_calls: Optional[List[Union[ToolCall, InvalidToolCall]]] = None
+    """If provided, tool calls associated with the message."""
+
     type: Literal["ai"] = "ai"
 
     @classmethod
     def get_lc_namespace(cls) -> List[str]:
         """Get the namespace of the langchain object."""
         return ["langchain", "schema", "messages"]
 
+    @root_validator
+    def _backwards_compat_tool_calls(cls, values: dict) -> dict:
+        raw_tool_calls = values.get("additional_kwargs", {}).get("tool_calls")
+        tool_calls = values.get("tool_calls") or values.get("tool_call_chunks")
+        if raw_tool_calls and not tool_calls:
+            warnings.warn(
+                "New langchain packages are available that more efficiently handle "
+                "tool calling. Please upgrade your packages to versions that set "
+                "message tool calls. e.g., `pip install --upgrade langchain-anthropic"
+                "`, pip install--upgrade langchain-openai`, etc."
+            )
+        try:
+            if issubclass(cls, AIMessageChunk):  # type: ignore
+                values["tool_call_chunks"] = default_tool_chunk_parser(raw_tool_calls)
+            else:
+                values["tool_calls"] = default_tool_parser(raw_tool_calls)
```

Comment:
Should this check to see if `tool_calls` is passed in on `values`?

Reply:
good catch!